cluster/src/main/java/io/scalecube/cluster/fdetector/FailureDetector.java

```diff
@@ -309,37 +290,28 @@ private void declareSuspected(Address member) {
    */
   private void declareTrusted(Address member) {
     if (suspectedMembers.remove(member)) {
-      LOGGER.debug("Member {} became TRUSTED", member);
-      subject.onNext(trusted(member));
+      LOGGER.debug("Member {} detected as TRUSTED by {}", member, transport.address());
+      subject.onNext(new FailureDetectorEvent(member, ClusterMemberStatus.TRUSTED));
     }
   }
 
-  private Address selectPingMember(List<Address> members) {
-    if (pingMember != null) {
-      return pingMember;
-    }
-    return members.isEmpty() ? null : selectRandomMembers(members, 1, null).get(0);
+  private Address selectPingMember() {
+    // TODO [AK]: Select ping member not randomly, but sequentially with shuffle after last member selected
+    return !members.isEmpty() ? members.get(ThreadLocalRandom.current().nextInt(members.size())) : null;
   }
 
```

Comment:
select ping member should be actually not random, as I wrote in TODO - in order to have stable upper bound on failure detection time.

And then it is not just selectRandomMembers, but has specific meaning as part of the algorithm (selecting ping request members).